docs/celery.rst

```diff
@@ -0,0 +1,119 @@
+    case_import_queue,,seconds,,Run case imports,minutes / hours,
+    email_queue,yes,seconds,"generally seconds, since people often blocked on receiving the email (registration workflows for example)",Send emails.,seconds,
+    export_download_queue,,seconds,seconds / minutes,Used for manually-triggered exports,minutes,
+    icds_dashboard_reports_queue,,seconds,fast,,,
+    background_queue,,,,,varies wildly,
+    beat,N/A,,,,,
+    case_rule_queue,,,,Run case update rules. Triggered by schedule,minutes / hours,
+    celery,,,,,,
+    celery_periodic,,,,,"Invoice generation: ~2 hours on production.  Runs as a single task, once per month.","I think this is one of the trickiest ones (and most heterogenous) because we run lots of scheduled tasks, that we expect to happen at a certain time, some of which we want at exactly that time and some we are ok with delay in start."
+    flower,N/A,,,,,
+    icds_aggregation_queue,yes,,initial task is immediate. follow up tasks are constrained by performance of previous tasks. recommend not tracking,Run aggregation tasks for ICDS. Triggered by schedule.,,
+    ils_gateway_sms_queue,,,,Custom queue for sending SMS for ILS Gateway project,,
+    logistics_background_queue,,,,Custom queue,,
+    logistics_reminder_queue,,,,Custom queue,,
+    saved_exports_queue,,,,Used only for regularly scheduled exports. Triggered by schedule.,minutes,"This queue is used only for regularly scheduled exports, which are not user-triggered. The time taken to process a saved export depends on the export itself. We now save the time taken to run the saved export as last_build_duration which can be used to monitor or move the task to a different queue that handles big tasks. Since all exports are triggered at the same time (midnight UTC) the queue gets big. Could be useful to spread these out so that the exports are generated at midnight in the TZ of the domain (see callcenter tasks for where this is already done)"
+    ucr_indicator_queue,no,,,Used for ICDS very expensive UCRs to aggregate,,
+    ucr_queue,no,,,Used to rebuild UCRs,minutes to hours,"This is where UCR data source rebuilds occur. Those have an extremely large variation. May be best to split those tasks like ""Process 1000 forms/cases, then requeue"" so as to not block"
+
+
+
+Soil
+====
+
+Soil is a Dimagi utility to provide downloads that are backed by celery.
+
+To use soil:
+
+.. code-block:: python
+
+    from soil import DownloadBase
+    from soil.progress import update_task_state
+    from soil.util import expose_cached_download
+
+    @task
+    def my_cool_task():
+        DownloadBase.set_progress(my_cool_task, 0, 100)
+
+        # do some stuff
+
+        DownloadBase.set_progress(my_cool_task, 50, 100)
+
+        # do some more stuff
+
+        DownloadBase.set_progress(my_cool_task, 100, 100)
+
+        expose_cached_download(payload, expiry, file_extension)
```

Comment:
can actually assign payload, expiry and file_extension to something here as an example so this becomes a standalone clean example so that one does not need to look at `expose_cached_download` to view the purpose of those.